lib/collection/src/shards/replica_set/update.rs

```diff
@@ -220,6 +224,33 @@ impl ShardReplicaSet {
 
         let (successes, failures): (Vec<_>, Vec<_>) = all_res.into_iter().partition_result();
 
+        // Advance clock if some replica echoed *newer* tick
+
+        let new_clock_tick = successes
+            .iter()
+            .filter_map(|(_, result)| {
+                let echo_tag = result.clock_tag?;
+
+                if echo_tag.peer_id != clock_tag.peer_id {
+                    // TODO: `log::warn`!?
+                    return None;
+                }
+
+                if echo_tag.clock_id != clock_tag.clock_id {
+                    // TODO: `log::warn`!?
+                    return None;
+                }
+
+                Some(echo_tag.clock_tick)
+            })
+            .max();
+
+        if let Some(new_clock_tick) = new_clock_tick {
+            if new_clock_tick > current_clock_tick || new_clock_tick == 0 {
```

Comment:
Shouldn't the second part be our old number?

```suggestion
            if new_clock_tick > current_clock_tick || clock_tag.clock_tick == 0 {
```

Reply:
In fact, can't we just totally remove this if-statement, and always `advance_to` the `new_clock_tick` value.

Reply:
Have to agree. Removed.